### PR TITLE
Allows users to connect google cloud storage integration from UI

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR allows users to connect a GCS integration.

## Related issue number (if any)
ENG 1597

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

DEMO: https://www.loom.com/share/b9a5e16a74bf41cf85d6d0e7d98bb48d

### CODE REVIEW

File: src/ui/common/src/components/integrations/dialogs/dialog.tsx [hunk 1]
```diff
@@ -29,6 +29,8 @@ import {
 } from '../../../utils/integrations';
 import { AirflowDialog } from './airflowDialog';
 import { BigQueryDialog } from './bigqueryDialog';
+import { CSVDialog } from './csvDialog';
```
Comment (saurav-c) [inline src/ui/common/src/components/integrations/dialogs/dialog.tsx:32]: Remove the CSV part

File: src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        spellCheck={false}
+        required={true}
+        label="Bucket*"
+        description="The name of the GCS bucket."
+        placeholder={Placeholders.bucket}
+        onChange={(event) => setBucket(event.target.value)}
+        value={bucket}
+      />
+
+      <IntegrationTextInputField
```
Comment (saurav-c) [inline src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx:44]: @vsreekanti Currently the only way to provide credentials is via a filepath. Do you think we should prioritize allowing users to upload a file as well (the same we do for connecting to s3)?

File: src/ui/common/src/utils/integrations.ts [hunk 3]
```diff
@@ -311,6 +319,11 @@ export const SupportedIntegrations: ServiceInfoMap = {
     activated: true,
     category: 'data',
   },
+  ['GCS']: {
+    logo: 'https://spiral-public-assets-bucket.s3.us-east-2.amazonaws.com/webapp/pages/integrations/google-cloud-storage.png',
```
Comment (saurav-c) [inline src/ui/common/src/utils/integrations.ts:323]: Fix this link

**Review by hsubbaraj-spiral — APPROVED** [review]

File: src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  credentials_path: '',
+  use_as_storage: '',
+};
+
+type Props = {
+  setDialogConfig: (config: IntegrationConfig) => void;
+};
+
+export const GCSDialog: React.FC<Props> = ({ setDialogConfig }) => {
+  const [bucket, setBucket] = useState<string>(null);
```
Comment (likawind) [inline src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx:19]: I just merged #320 and you might want to rebase to follow the new pattern. You could refer to any other dialog like snowflake as example. Happy to meet offline to help!

**Review by likawind — COMMENTED** [review]
Looks good! Only thing is to check merge conflicts :)

File: src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx (hunk 2 above)
Comment (saurav-c) [reply src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx]: Added the file upload method as we discussed

File: src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx (hunk 4 above)
Comment (saurav-c) [reply src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx]: Rebased! The new method is cool :)

**Pull request merged**